tests/main/document-portal-activation/task.yaml

```diff
@@ -17,75 +17,75 @@ description: |
 # Disabled on Ubuntu Core because it doesn't provide the "desktop"
 # slot, and Amazon Linux because it doesn't have the required Python 3
 # packages to run the test.
-systems: [ -ubuntu-core-*, -amazon-linux-2-*, -centos-* ]
-
-environment:
-    XDG_RUNTIME_DIR: /run/user/$(id -u)
+systems:
+    - -amazon-linux-2-*
+    - -centos-*
+    - -ubuntu-14.04-*
+    - -ubuntu-core-*
 
 prepare: |
-    #shellcheck source=tests/lib/pkgdb.sh
-    . "$TESTSLIB"/pkgdb.sh
     #shellcheck source=tests/lib/snaps.sh
     . "$TESTSLIB"/snaps.sh
     install_local test-snapd-desktop
     install_local test-snapd-sh
     rm -f /usr/share/dbus-1/services/fake-document-portal.service
-    mkdir -p "$XDG_RUNTIME_DIR"
-    rm -rf "${XDG_RUNTIME_DIR:?}/*" "${XDG_RUNTIME_DIR:?}/.[!.]*"
+    session-tool --prepare -u test
+
+    # The fake portal writes to current working directory so let's give it permission to.
+    touch doc-portal.log
+    touch report-error.txt
+    chown test doc-portal.log
+    chown test report-error.txt
 
 restore: |
-    #shellcheck source=tests/lib/pkgdb.sh
-    . "$TESTSLIB"/pkgdb.sh
-    [ -f dbus-launch.pid ] && kill "$(cat dbus-launch.pid)"
     rm -f /usr/share/dbus-1/services/fake-document-portal.service
-    rm -rf "${XDG_RUNTIME_DIR:?}/*" "${XDG_RUNTIME_DIR:?}/.[!.]*"
+    rm -f doc-portal.log
+    rm -f report-error.txt
+    session-tool --restore -u test
 
 execute: |
     #shellcheck source=tests/lib/systems.sh
     . "$TESTSLIB"/systems.sh
 
-    check_stderr() {
-        test -n "$1"
-        if is_cgroupv2; then
-            [ "$(cat "$1")" = "WARNING: cgroup v2 is not fully supported yet, proceeding with partial confinement" ]
-        else
-            [ "$(wc -c < "$1")" -eq 0 ]
-        fi
-    }
-    echo "No output on stderr when running without a session bus"
-    user_data="$HOME/snap/test-snapd-desktop/current"
-    test-snapd-desktop.check-dirs "$user_data" 2>stderr.log
-    check_stderr stderr.log
-
-    echo "Starting session bus"
-    eval "$(dbus-launch --sh-syntax)"
-    echo "$DBUS_SESSION_BUS_PID" > dbus-launch.pid
+    if is_cgroupv2; then
+        check_stderr() {
+            test -n "$1"
+            MATCH "WARNING: cgroup v2 is not fully supported yet, proceeding with partial confinement" < "$1"
```

Comment:
I think this is incorrect compared to the previous code. The removed code checked that the warning is the only content of the stderr file. In the current code the check is still true when the warning is followed by some random content.

Reply:
Ah, good point. I'll adjust this.

Reply:
Done